### ISSUE TITLE
Dynamic field creation using value of other field 

### ISSUE THREAD

**Issue by prashanttct07** [opensearch-project/data-prepper#4176]
**Is your feature request related to a problem? Please describe.**
We would like to have custom field created with the value of other field. For ex: Below is the sample doc we have: 

```
"_source": {
    "name": "system.cpu.used.pct",
    "value": 0.767015
  }

```

Where name has high cardinal value and can have hundreds of unique value such as memory, disk etc

**Describe the solution you'd like**
As a solution we would like a new fields to be created where key would be value from name, and value would be value from value. 

So, output would be like:
```
"_source": {
    "system.cpu.used.pct": 0.767015
    "name": "system.cpu.used.pct",
    "value": 0.767015
  }

```
Additionally it would be good if we have an option where you can replace . via _ as well. For ex: 

```
"_source": {
    "system.cpu.used.pct": 0.767015,
    "system_cpu_used_pct": 0.767015,
    "name": "system.cpu.used.pct",
    "value": 0.767015
  }
```

**Comment by oeyh** [comment]
Will resolve it through issue #4175.

**Comment by oeyh** [comment]
Regarding the request to "replace . via _", we can do it using the substitute_string processor before using it as field key:
```
  processor:
    - substitute_string:
        entries:
          - source: "name"
            from: "."
            to: "_"

```

**Comment by oeyh** [comment]
Closed via #4182
